File: src/0094.binary-tree-inorder-traversal.94/solution.go
package binarytreeinordertraversal94

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}
	result := inorderTraversal(root.Left)
	result = append(result, root.Val)
	result = append(result, inorderTraversal(root.Right)...)
	return result
}

File: src/0094.binary-tree-inorder-traversal.94/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> list = new ArrayList<>();
        if (root != null) {
            _inorderTraversal(root, list);
        }
        return list;
    }

    private void _inorderTraversal(TreeNode root, List<Integer> list) {
        if (root.left != null) {
            _inorderTraversal(root.left, list);
        }

        list.add(root.val);

        if (root.right != null) {
            _inorderTraversal(root.right, list);
        }

    }

}